caf-audit/src/main/java/com/hpe/caf/auditing/elastic/ElasticAuditTransportClientFactory.java

```diff
@@ -49,49 +50,63 @@ private ElasticAuditTransportClientFactory() {
      * @return TransportClient
      * @throws ConfigurationException exception thrown if host is unknown
      */
-    public static TransportClient getTransportClient(String hostAndPortValues, String clusterName) throws ConfigurationException{
+    public static TransportClient getTransportClient(String hostAndPortValues, String clusterName) throws ConfigurationException {
         final TransportClient transportClient;
 
-        try {
-            Settings settings = Settings.builder()
-                    .put("cluster.name", clusterName).build();
-            transportClient = new PreBuiltTransportClient(settings);
+        if (hostAndPortValues != null && !hostAndPortValues.isEmpty()) {
+            //  Split comma separated list of ES hostname and port values.
+            final String[] hostAndPortArray = hostAndPortValues.split(",");
 
-            if (hostAndPortValues != null && !hostAndPortValues.isEmpty()) {
-                //  Split comma separated list of ES hostname and port values.
-                final String[] hostAndPortArray = hostAndPortValues.split(",");
+            // If there is more than one Elasticsearch host then set the client.transport.sniff to true as we are dealing with a cluster
+            Settings.Builder settingsBuilder = Settings.builder()
+                    .put("cluster.name", clusterName);
+            if (hostAndPortArray.length > 1) {
+                settingsBuilder.put("client.transport.sniff", true);
```

Comment:
Chris what are the benefits of this setting? Is this set in case additional nodes are added to the Elasticsearch cluster which have not been specified in the hostAndPortValues configuration? Thanks.